craft_parts/plugins/base.py

```diff
@@ -132,7 +132,7 @@ class PluginModel(PluginPropertiesModel):
 
 
 def extract_plugin_properties(
-    data: Dict[str, Any], *, plugin_name: str, required: Optional[List[str]] = None
+    data: Dict[str, Any], *, plugin_name: str, required: Collection[str] = ()
```

Comment:
Nice!  This is so much better than the old tricks we had to use when we wanted to use a mutable default value.